sorc/build_ufs.sh

```diff
@@ -0,0 +1,49 @@
+#! /usr/bin/env bash
+set -eux
+
+# Build ATMW by default
+APP="ATMW"
+CCPP_SUITES="FV3_GFS_v16,FV3_GFS_v16_RRTMGP,FV3_GFS_v16_ugwpv1"
+
+while getopts "ac" option; do
+  case "${option}" in
+    a)
+      APP="ATMAERO"
+      CCPP_SUITES="FV3_GFS_v16,FV3_GFS_v16_ugwpv1"
+      shift
+      ;;
+    c)
+      APP="S2SW"
+      CCPP_SUITES="FV3_GFS_v16_coupled,FV3_GFS_v16_couplednsst,FV3_GFS_v16_coupled_nsstNoahmpUGWPv1"
+      shift
+      ;;
+    *)
+      echo "Unrecognized option: ${1}"
+      exit 1
+      ;;
+  esac
+done
+
+source ./machine-setup.sh > /dev/null 2>&1
+cwd=$(pwd)
+
+# Remove previous build directory if it exists
+if [ -d ufs_model.fd/build ]; then
+  rm -R ufs_model.fd/build
+fi
+
+# Set target platform
+case "${target}" in
+  hera|orion|stampede)
+    target=${target}.intel
+    ;;
+esac
+
+MOD_PATH=$cwd/ufs_model.fd/modulefiles
+
+module purge
+
+cd ufs_model.fd/
+module use ${MOD_PATH}
+module load ufs_${target}
+CMAKE_FLAGS="-DAPP=${APP} -DCCPP_SUITES=${CCPP_SUITES}" ./build.sh
```

Comment:
Please don't use the `build.sh` in the UFS weather model.  Its use is for the regression tests.  Users and developers should use `cmake` instructions required to build the model.

```suggestion
mkdir -p build && cd build
cmake -DAPP=${APP} -DCCPP_SUITES=${CCPP_SUITES} ..
OMP_NUM_THREADS=1 make -j ${BUILD_JOBS:-8} VERBOSE=${BUILD_VERBOSE:-} 
```

Reply:
I am fine making this change, but until now this is how we've been building UFS in both branches.

Reply:
Sure. 
Going forward, if we want to maintain portability.  We should:
- ensure we use the build instructions and not the build script provided by the components
- minimize reliance on modulefiles (this is impossible on supported platforms).  But, if we use build instructions, we take the module load environment out of the equation, since the module files provide paths to the third party libraries.  So, if we land on a machine with no modules but the build environment in the `PATH`, we can build without modulefiles and the component build scripts.

Reply:
Updated to use cmake directly. Per conversation, postponed moving build directory out of UFS repo. A future PR will do that for all components.